apm-sniffer/apm-sdk-plugin/spring-plugins/spring-rmi-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/rmi/RmiClientInterceptorDoInvokeInterceptor.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.spring.rmi;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+import org.springframework.remoting.rmi.RmiClientInterceptor;
+
+public class RmiClientInterceptorDoInvokeInterceptor implements InstanceMethodsAroundInterceptor {
+    public static final String OPERATE_NAME_PREFIX = "SpringRmi/client/";
+
+    @Override
+    public void beforeMethod(final EnhancedInstance objInst, final Method method, final Object[] allArguments,
+                             final Class<?>[] argumentsTypes, final MethodInterceptResult result) throws Throwable {
+
+        RmiClientInterceptor interceptor = (RmiClientInterceptor) objInst;
+        URL url = new URL(null, interceptor.getServiceUrl(), new DummyURLStreamHandler());
+        String name = url.getPath();
+        if (name != null && name.startsWith("/")) {
+            name = name.substring(1);
+        }
+
+        AbstractSpan span = ContextManager.createExitSpan(OPERATE_NAME_PREFIX + name, url.getHost() + ":" + url.getPort());
```

Comment:
Where is Context#Inject?

Reply:
Java Rmi I took a look and did not support header, so I did not inject.

Reply:
If it didn't allow inject header, we don't need to provide plugin. Because the trace would still break.

Reply:
When you face the unnormal case, please start a discussion rather than writing codes. This seems wasting your limited time.

Reply:
> When you face the unnormal case, please start a discussion rather than writing codes. This seems wasting your limited time.

ok i close.